Enforce UTF-8 charset for core controllers

wouterj opened the pull request:
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #7532 |
| License | MIT |
| Doc PR | - |

See https://github.com/symfony/symfony/issues/7617 and https://github.com/symfony/symfony/pull/7532 for the context of this PR.

wouterj committed:
Enforce UTF-8 charset for core controllers

nicolas-grekas commented:
:+1:

nicolas-grekas commented:
Maybe a few assertions added ProfilerControllerTest could help prevent breaking the fix

GromNaN commented:
You can use the syntax Response::create() to chain the setCharset.

WouterJ commented:
Thanks for the great suggestions, @nicolas-grekas and @GromNaN! I've now updated the PR to use `Request::create()` and I've added an assertion to the `ProfilerControllerTest` and `ExceptionControllerTest`.

javiereguiluz commented on src/Symfony/Bundle/TwigBundle/Tests/Controller/ExceptionControllerTest.php:
```diff
@@ -39,6 +39,7 @@ public function testOnlyClearOwnOutputBuffers()
         $controller = new ExceptionController($twig, false);
-        $controller->showAction($request, $flatten);
+        $response = $controller->showAction($request, $flatten);
+        $this->assertEquals('UTF-8', $response->getCharset(), 'Request charset is explicitely set to UTF-8');
```
typo: `explicitely` -> `explicitly`

javiereguiluz commented on src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php:
```diff
@@ -69,6 +69,7 @@ public function testReturns404onTokenNotFound()
 
         $response = $controller->toolbarAction(Request::create('/_wdt/found'), 'found');
         $this->assertEquals(200, $response->getStatusCode());
+        $this->assertEquals('UTF-8', $response->getCharset(), 'Request charset is explicitely set to UTF-8');
```
typo: `explicitely` -> `explicitly`

WouterJ commented:
Fixed the typos

fabpot commented:
Thank you @WouterJ.
